x/metadata/keeper/mocks_test.go

```diff
@@ -325,14 +332,327 @@ func (a *MockAuthorization) Reset() {
 
 // String returns a string of this MockAuthorization and satisfies the authz.Authorization interface.
 func (a *MockAuthorization) String() string {
-	_ = MockAuthorization{
-		Name:              "",
-		AcceptResponse:    authz.AcceptResponse{},
-		AcceptResponseErr: nil,
-		AcceptCalls:       nil,
-	}
 	return fmt.Sprintf("MockAuthorization{%q, %v, %v, %d}", a.Name, a.AcceptResponse, a.AcceptResponseErr, a.AcceptCalls)
 }
 
 // ProtoMessage satisfies the authz.Authorization interface.
 func (a *MockAuthorization) ProtoMessage() {}
+
+var _ keeper.MarkerKeeper = (*MockMarkerKeeper)(nil)
+
+// MockMarkerKeeper is a mocked keeper.MarkerKeeper.
+type MockMarkerKeeper struct {
+	IsMarkerAccountResults map[string]bool
+	IsMarkerAccountCalls   []sdk.AccAddress
+}
+
+func NewMockMarkerKeeper() *MockMarkerKeeper {
+	return &MockMarkerKeeper{
+		IsMarkerAccountResults: make(map[string]bool),
+	}
+}
+
+// WithIsMarkerAccountResults sets up the provided addrs to return true from IsMarkerAccount.
+func (k *MockMarkerKeeper) WithIsMarkerAccountResults(addrs ...sdk.AccAddress) *MockMarkerKeeper {
+	for _, addr := range addrs {
+		k.IsMarkerAccountResults[string(addr)] = true
+	}
```

Comment:
I pointed this out in another comment too. The `sdk.AccAddress.String()` method is more expensive than casting to a string. An `AccAddress` cannot be used as the key for a `map` directly since it's a slice, so when I need to use an `AccAddress` as a key for a `map`, I need to first convert it to a string somehow. Since that string is only being used as keys in a map, it doesn't really matter if it has unprintable characters, and casting to a string is significantly cheaper than calling `.String()`, so I chose to use casting here.